builtin/credential/approle/path_login.go

```diff
@@ -219,17 +220,30 @@ func (b *backend) pathLoginUpdate(ctx context.Context, req *logical.Request, dat
 		metadata = entry.Metadata
 	}
 
-	if len(role.BoundCIDRList) != 0 {
+	if len(role.SecretIDBoundCIDRs)+len(role.TokenBoundCIDRs) != 0 {
 		if req.Connection == nil || req.Connection.RemoteAddr == "" {
 			return nil, fmt.Errorf("failed to get connection information")
 		}
-
-		belongs, err := cidrutil.IPBelongsToCIDRBlocksSlice(req.Connection.RemoteAddr, role.BoundCIDRList)
-		if err != nil || !belongs {
-			return logical.ErrorResponse(errwrap.Wrapf(fmt.Sprintf("source address %q unauthorized by CIDR restrictions on the role: {{err}}", req.Connection.RemoteAddr), err).Error()), nil
+		if len(role.SecretIDBoundCIDRs) != 0 {
+			belongs, err := cidrutil.IPBelongsToCIDRBlocksSlice(req.Connection.RemoteAddr, role.SecretIDBoundCIDRs)
+			if err != nil || !belongs {
+				return logical.ErrorResponse(errwrap.Wrapf(fmt.Sprintf("source address %q unauthorized by CIDR restrictions on the role: {{err}}", req.Connection.RemoteAddr), err).Error()), nil
+			}
+		}
+		if len(role.TokenBoundCIDRs) != 0 {
```

Comment:
We shouldn't be doing this check -- login should check against secret id bound CIDRs only, not both.

Reply:
:heavy_check_mark: